cmake/ports/node/portfile.cmake

```
@@ -26,12 +26,55 @@ elseif (APPLE)
 else ()
     # else Linux desktop
     if (VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
-        vcpkg_download_distfile(
-            NODE_SOURCE_ARCHIVE
-            URLS "${EXTERNAL_BUILD_ASSETS}/dependencies/node/node-install-18.14.2-ubuntu-18.04-amd64-release.tar.xz"
-            SHA512 ff5ca5c27b811d20ac524346ee122bcd72e9e85c6de6f4799f620bb95dac959ce910cc5bb2162ed741a7f65043aa78173ecd2ce5b92f5a4d91ecb07ce71fa560
-            FILENAME node-install-18.14.2-ubuntu-18.04-amd64-release.tar.xz
+        vcpkg_from_github(
+            OUT_SOURCE_PATH SOURCE_PATH
+            REPO nodejs/node
+            REF v18.16.0
+            SHA512 9b983b899acd02e7ed761bc3633fc56855e10335fcdb558a29d1cf068ce1125991c9a781616d82a9dc90be6e8ba1bf4a34a10a92c6b7db9cbe33ef7fa7dda67f
+            HEAD_REF v18.16.0
+        )
+        # v18.16.0-e8bd828d3a.clean is getting added by VCPKG. It will need to be updated when we update libnode version.
+        file(COPY ${SOURCE_PATH} DESTINATION "${CURRENT_BUILDTREES_DIR}")
```

```suggestion
        file(COPY ${SOURCE_PATH}/* DESTINATION "${CURRENT_BUILDTREES_DIR}")
```
I wonder if the SOURCE_PATH is "v18.16.0-e8bd828d3a.clean" and we copy it *into* CURRENT_BUILDTREES_DIR.
The CMake documentation suggests that SOURCE_PATH is interpreted as a file that is copied into the CURRENT_BUILDTREES_DIR directory.
So if we give it all files in SOURCE_PATH instead of SOURCE_PATH itself, maybe we won't have to deal with "v18.16.0-e8bd828d3a.clean" later.

---

Turns out that the documentation doesn't just suggest it, but means it.